### PR TITLE
[Serve][Doc] Add back reconfigure documentation 

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
The documentation about `reconfigure` was removed during the 2.x doc refactor. This PR adds it back to the right location (config file). 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/production-guide/config.md [hunk 1]
```diff
@@ -170,3 +170,31 @@ Serve will set `num_replicas=5`, using the config file value, and `max_concurren
...
 :::
+
+## Dynamically adjusting parameters in deployment
+
+The `user_config` field can be used to supply structured configuration for your deployment. You can pass arbitrary JSON serializable objects to the YAML configuration. Serve will then apply it to all running and future deployment replicas. The application of user configuration *will not* restart the replica. This means you can use this field to dynamically:
+- adjust model weights and versions without restarting the cluster.
+- adjust traffic splitting percentage for your model composition graph.
+- configure any feature flag, A/B tests, and hyper-parameters for your deployments.
+
+The enable your deployment supports the user configuration feature, you just need to implement a `reconfigure` method that takes one argument:
```
Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/config.md:181]:
```suggestion
To enable the `user_config` feature, you need to implement a `reconfigure` method that takes a dictionary as its only argument:
```

File: doc/source/serve/production-guide/config.md [hunk 2]
```diff
@@ -170,3 +170,31 @@ Serve will set `num_replicas=5`, using the config file value, and `max_concurren
...
+
+```python
+@serve.deployment
+class Model:
+    def reconfigure(self, config: Dict[str, Any]):
+        self.threshold = config["threshold"]
+```
+
+The reconfigure method is called when the class is created if user_config is set. In particular, it's also called when new replicas are created in the future if scale up your deployment later. The reconfigure method is also called each time user_config is updated.
```
Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/config.md:190]:
```suggestion
If the `user_config` is set when the deployment is created (e.g. in the decorator or the Serve config file), this `reconfigure` method is called right after the deployment's `__init__` method, and the `user_config` is passed in as an argument. You can also trigger the `reconfigure` method by updating your Serve config file with a new `user_config` and reapplying it to your Ray cluster.
```

**Review by shrekris-anyscale — COMMENTED** [review]
Nice work! I added a couple suggestions.

**Review by architkulkarni — APPROVED** [review]
Good catch!

**Pull request merged**